README.md

```diff
@@ -1,24 +1,220 @@
 
-* System dependencies
+<!-- PROJECT DESCRIPTION -->
 
-* Configuration
+# 📖 Hotel Hub (Backend) <a id="about-project"></a>
 
-* Database creation
+**Hotel Hub** is a website that offers hotel booking and reservation services for travelers, providing a convenient way to find and book accommodations. You can select available rooms for a graphic list on the home page of the website, or you can just click on the reserve menu item and see the full list of available rooms. You can find the repository for the [front end](https://github.com/ProgramKingAbel/hotel-hub-frontend.git) of this project here.
 
-* Database initialization
+Backend is implement as API-only. The front end is implemented utilizing the React framework.
 
-* How to run the test suite
+-  The backend Rails app is implemented as API-only. The application starts by fetching the `users` method on the backend, if the backend is available then the user will login by making a query to `users/new_session`.
 
-* Services (job queues, cache servers, search engines, etc.)
+-  The room controller provides an `index` method to get all the data for rooms, and a `show` method to get details about a specific room. The `destroy` method deactivates the room when the user sends a delete query by clicking on delete.
 
-* Deployment instructions
+-  The reservation controller also provides `index` methods. Index is used to get a list of all reservations made. 
+ 
+## 🛠 Built With <a id="built-with"></a>
 
-* ...
+### Tech Stack <a id="tech-stack"></a>
+
+<details>
+  <summary>Ruby on Rails</summary>
+  <ul>
+    <li><a>https://rubyonrails.org/</a></li>   
+  </ul>
+</details>
+
+<!-- Features -->
+
+### Key Features <a id="key-features"></a>
+
+- **Ability to see the featured rooms on the homepage**
+- **The ability to reserve rooms based on selecting the date and room name**
+- **The ability to add rooms to the database**
+- **The ability to delete rooms**
+
+<p align="right">(<a href="#readme-top">back to top</a>)</p>
+
+## 🚀 Live Demo <a id="live-demo"></a>
+
+Live demo for the website [Coming Soon]().
+
+<p align="right">(<a href="#readme-top">back to top</a>)</p>
+
+## 🎬 Kanban Board <a id="kanban"></a>
+
+Link to the [Kanban Board](https://github.com/ProgramKingAbel/hotel-hub-backend/projects/1)
+
+All 5 authors have been contributing to this project from start to finish.
+
+
+<!-- GETTING STARTED -->
+
+## 💻 Getting Started <a id="getting-started"></a>
+
+To get a local copy up and running, follow these steps.
+
+### Prerequisites
+
+In order to run this project, you need to:
+
+- Install Ruby
+- Install Rails
+- Install Posgresql
+<p align="right">(<a href="#readme-top">back to top</a>)</p>
+
+### Setup
+
+Clone this repository to your desired folder:
+
+``` 
+  git clone https://github.com/ProgramKingAbel/hotel-hub-backend.git 
+```
+
+### Install
+
+Install this project with these commands:
+
+```
+  cd hotel-hub
+  bundle install
+  rails db:create
+  rails db:migrate
```

Comment:
- [ ] The last migration is failing:

![image](https://github.com/ProgramKingAbel/hotel-hub-backend/assets/57421823/ce1f2218-9ea0-4b10-a0e4-161364bc72f2)

This is because this migration is trying to delete certain columns in the Users table that don't exist:

![image](https://github.com/ProgramKingAbel/hotel-hub-backend/assets/57421823/b7574662-e81f-449e-bc23-2fbb487cfb43)

## Tips and extra help

Use this command to rollback all your migrations from the database:

```rake db:migrate VERSION=0```

After that command, you can delete and/or correct the last migration, which is failing.

Now you can run your migrations again.

Kindly, correct your migrations so they succeed.

_note: I'll do this myself locally in order to be able to continue the review._